doc/contributing/architecture.md

```diff
@@ -0,0 +1,107 @@
+TerriaJS is usually used in a single-page web application to provide mapping and catalog exploration capabilities.  It is written in ECMAScript (JavaScript) 2015+ and compiled to ECMAScript 5 in order to run in any relatively modern web browser, including Internet Explorer 9 or later.  Some features, such as 3D and vector tile region mapping, require more recent browsers, but TerriaJS can be expected to degrade gracefully in older browsers by disabling unsupported features or informing the user of the limitations of their browser. 
+
+## Layers
+
+TerriaJS has a number of subdirectories of the `lib` directory, each of which corresponds to a conceptual layer in the design of TerriaJS.  The layers are:
+
+| Name | Purpose |
+|------|---------|
+| ThirdParty | This directory contains third-party code that is not available via npm packages for whatever reason. |
+| Core | Low-level utility classes and functions that don't depend on a UI toolkit (i.e. React) or a mapping library (i.e. Cesium or Leaflet). |
+| Map | Classes and functions to work with or extend a mapping library (i.e. Cesium or Leaflet). This layer should not depend on a particular UI toolkit (i.e. React). |
+| Charts | Similar to the `Map` layer, except for charting instead of mapping.  This depends on the charting library (D3) but not on React. |
+| Models | This is heart of TerriaJS.  Classes and functions in this layer know how to interface with catalog servers, map servers, and geospatial data formats.  Most of the logic of a TerriaJS application is found in this layer.  However, it does _not_ depend on React. |
+| ViewModels and ReactViewModels | Contains classes and functions that are closely associated with the UI but cannot be described as React components or views.  These two directories are separate mostly for historical reasons; there is little practical difference between them. |
+| ReactViews | This is the React-based user interface of TerriaJS.  All of the React components are found in this layer. |
+
+## User interface
+
+TerriaJS uses the [React](https://reactjs.org/) library as the basis for its user interface.  The top-level entry point is [StandardUserInterface.jsx](https://github.com/TerriaJS/terriajs/blob/master/lib/ReactViews/StandardUserInterface/StandardUserInterface.jsx).  It is expected that most applications will use this standard user interface, but a sophisticated TerriaJS application could use its own version instead in order to allow extreme customization of the interface.
+
+For styling, we use [Sass](http://sass-lang.com/) in CSS modules loaded via the Webpack `css-loader`, meaning that each React component imports its own local CSS styles from a `.scss` file.
```

Comment:
Might want to mention that it's CSS Modules here, not just CSS Loader.

Reply:
It says "CSS modules loaded via the Webpack css-loader".  Is there more to it than that?  Admittedly I only barely understand how the CSS works in TerriaJS. :)